app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -180,27 +180,21 @@ class ReviewRegistrationController @Inject() (
         registration.organisationDetails.incorporationDetails.flatMap(
           details => details.registration.registeredBusinessPartnerId
         )
-    }.getOrElse(throw new Exception("Safe Id is required for a Subscription create"))
+    }.getOrElse(throw new IllegalStateException("Safe Id is required for a Subscription create"))
 
   private def getSoleTraderDetails()(implicit
     request: JourneyRequest[AnyContent]
   ): Future[SoleTraderIncorporationDetails] =
-    request.registration.organisationDetails.soleTraderDetails.fold(
-      throw new Exception("Unable to fetch sole trader details from cache")
-    )(details => Future.successful(details))
+    Future.successful(request.registration.organisationDetails.soleTraderDetails.get)
```

Comment:
Is there a reason why the exception handling is removed?

Reply:
Yes, although possibly not the best reason. I removed it mainly because it wasn't testable any more nor was it possible to occur due to changes to `Registration.isCheckAndSubmitReady()` although that is not readily apparent to the reader.